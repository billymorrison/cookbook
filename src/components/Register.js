import React, { useState } from 'react';
import { Link, useHistory } from "react-router-dom";
import default1 from "../Media/default1.png";
import axios from "axios";
import {
  StyledInput,
  StyledLabel,
  StyledForm,
  MainLoginArea,
  StyledImageSection,
  StyledButton,
} from "./GlobalStyles";

const Register = ({}) => {
  const [value, SetValue] = useState();

  const handleChange = (e) => {
    SetValue({
      ...value,
      [e.target.name]: e.target.value,
    });
  };

  const history = useHistory();
  const handleSubmit = (e) => {
    e.preventDefault();
    if (value.password === value.confirmPassword)
      console.log("Passwords match!");

    axios
      .post("http://localhost:3000/user/register", {
        email: value.email,
        password: value.password,
        displayName: value.displayName,
      })
      .then((res) => {
        history.push("/");
        console.log(res);
      })
      .catch((err) => console.error(err));
  };

  return (
    <MainLoginArea>
      <StyledForm action="submit" onSubmit={handleSubmit}>
        <h1>Register</h1>
        <StyledLabel>
          Email
          <StyledInput
            type="email"
            required
            onChange={handleChange}
            name="email"
          />
        </StyledLabel>
        <StyledLabel>
          Display Name
          <StyledInput
            type="text"
            required
            onChange={handleChange}
            name="displayName"
          />
        </StyledLabel>
        <StyledLabel>
          Password
          <StyledInput
            type="password"
            required
            onChange={handleChange}
            name="password"
          />
        </StyledLabel>
        <StyledLabel>
          Confirm Password
          <StyledInput
            type="password"
            required
            onChange={handleChange}
            name="confirmPassword"
          />
        </StyledLabel>
        <StyledButton type="submit">Submit</StyledButton>
        <Link to="/">Already a member? Login here.</Link>
      </StyledForm>
      <StyledImageSection src={default1} alt="food" />
    </MainLoginArea>
  );
};

export default Register;
