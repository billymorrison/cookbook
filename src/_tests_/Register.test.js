import ReactDOM from "react-dom";
import Register from "../components/Register";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

it("renders register component without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Router><Register /></Router>, div)
});

it("renders add recipe correctly", () => {
  const {getByTestId} = render(<Router><Register /></Router>);
  expect(getByTestId("register")).toHaveTextContent("Register");
})

it("matches snapshot", () => {
  const tree = renderer.create(<Router><Register /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
})