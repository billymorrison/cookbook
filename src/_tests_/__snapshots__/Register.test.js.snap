// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-dlfnbm lUGTh"
  data-testid="register"
>
  <form
    action="submit"
    className="sc-hKgILt icauyc"
    onSubmit={[Function]}
  >
    <h1>
      Register
    </h1>
    <label
      className="sc-gsTCUz fYmKmH"
    >
      Email
      <input
        className="sc-bdfBwQ bDhmHQ"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </label>
    <label
      className="sc-gsTCUz fYmKmH"
    >
      Display Name
      <input
        className="sc-bdfBwQ bDhmHQ"
        name="displayName"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label
      className="sc-gsTCUz fYmKmH"
    >
      Password
      <input
        className="sc-bdfBwQ bDhmHQ"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </label>
    <label
      className="sc-gsTCUz fYmKmH"
    >
      Confirm Password
      <input
        className="sc-bdfBwQ bDhmHQ"
        name="confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </label>
    <button
      className="sc-jSgupP iSLeyX"
      type="submit"
    >
      Submit
    </button>
    <a
      href="/"
      onClick={[Function]}
    >
      Already a member? Login here.
    </a>
  </form>
  <img
    alt="food"
    className="sc-gKsewC cvRtih"
    src="default1.png"
  />
</div>
`;
